graphql/rules.lua

```diff
@@ -516,6 +516,7 @@ local function isVariableTypesValid(argument, argumentType, context,
     if hasDefault and variableType.__type ~= 'NonNull' then
       variableType = types.nonNull(variableType)
     end
+    if argumentType.kind then argumentType = argumentType.kind end
```

Comment:
(And same here.)

Reply:
Since this PR tests run correct even without the fix, I separated it to https://github.com/tarantool/graphql/pull/31 with corresponding tests and descriptions.